core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -184,6 +184,48 @@ class PlaintextConsumerTest extends BaseConsumerTest {
   }
 
   @Test
+  def testSubsequentPatternSubscription() {
+    val numRecords = 10000
+    sendRecords(numRecords)
+
+    val fooTopic = "foo" // matches both subscription patterns
+    TestUtils.createTopic(this.zkUtils, fooTopic, 1, serverCount, this.servers)
+    sendRecords(1000, new TopicPartition(fooTopic, 0))
+
+    assertEquals(0, this.consumers.head.assignment().size)
+
+    val pattern1 = Pattern.compile(".*") // both 'topic' and 'foo' match this 
+    this.consumers.head.subscribe(pattern1, new TestConsumerReassignmentListener)
+    this.consumers.head.poll(50)
+
+    var subscriptions = Set(
+      new TopicPartition(topic, 0),
+      new TopicPartition(topic, 1),
+      new TopicPartition(fooTopic, 0))
+
+    TestUtils.waitUntilTrue(() => {
+      this.consumers.head.poll(50)
+      this.consumers.head.assignment() == subscriptions.asJava
+    }, s"Expected partitions ${subscriptions.asJava} but actually got ${this.consumers.head.assignment()}")
+
+    val pattern2 = Pattern.compile("f.*") // only 'foo' matches this
```

Comment:
@vahidhashemian The patch looks good, I just want to call out one thing that this test doesn't actually cover. We have other tests that validate that when we add topics, after a metadata refresh the new topic is included in the subscription when it matches the regex. This test, by checking for the removal of some topics, already validates that when metadata is updated we'll see the subscription updated to reflect the new regex. However, the *cause* of the metadata refresh may be either a) the resubscription or b) the normal metadata refresh interval. For (b), to make some other tests fast, the base class sets it to only 100ms. If we want this test to also validate that the resubscription forces an *immediate* metadata refresh and the subscription change is reflected promptly, I think we'd need to use a consumer with the metadata refresh interval overridden (see a few other tests in here that use custom settings for an example), add another topic before the resubscription (the only thing that will validate metadata changes), make sure it is included in the new regex, and then validate as we are already.

I think this is just some minor tweaking of the test, so hopefully straightforward to validate. A comment explaining what we're trying to make sure is happening would also be helpful as @hachikuji and I just had to spend some time thinking through what one of the other tests was actually evaluating.